tests/NestedSetsBehaviorTest.php

```diff
@@ -641,6 +641,29 @@ public function testIsLeaf()
         $this->assertFalse(Tree::findOne(1)->isLeaf());
     }
 
+    public function testMakeTreeWithTreeAttribute()
+    {
+        $dataSet = $this->createFlatXMLDataSet(__DIR__ . '/data/clean.xml');
+        $this->getDatabaseTester()->setDataSet($dataSet);
+        $this->getDatabaseTester()->onSetUp();
+
+        $node = new MultipleTree(['name' => 'Root', 'tree' => 10]);
+        $this->assertTrue($node->makeRoot());
+
+        $dataSet = $this->getConnection()->createDataSet(['multiple_tree']);
+        $expectedDataSet = $this->createFlatXMLDataSet(__DIR__ . '/data/test-make-tree-with-tree-attribute.xml');
+        $this->assertDataSetsEqual($expectedDataSet, $dataSet);
+    }
+
+    /**
+     * @expectedException \yii\db\Exception
+     */
+    public function testMakeRootNewExceptionIsRaisedWhenMultiTreeIsCreatedWithExistingTreeAttribute()
+    {
+        $node = new MultipleTree(['name' => 'Root 4', 'tree' => 1]);
```

Comment:
@creocoder Before merging, maybe you can explain why at this point there is no tree with tree Id 10 here. It is created in the previous test and the DB is not cleared as far as I can see. I'm still quite confused about the state of the DB for each test. I only came up with this solution by trial and error.